shell.go

```diff
@@ -7,6 +7,27 @@ import (
 	"unsafe"
 )
 
+// ShellError returns error information when shell commands do not succeed.
+type ShellError struct {
```

Comment:
So, there are 2 ways:
- the more idiomatic way was to return 2 errors: one for Windows and one for the Shell execution itself. That way, the caller can do an `errors.As` to know which kind of error it is.
- another way would be to do this and rely on the exit code. but grow a `.IsWindowsError()` method returning a bool.

Either way, I think we should differentiate those. Do you mind refreshing me with what we do for other Exec commands? We should align them all.

Whatever decision you take, I think this will need some dedicated tests.

Reply:
Hmm it'll be a bit tedious. Possible outputs:

- errorString: something failed before or at the syscall
    - distro not registered (or failed to access registry)
    - distroname cannot be converted to UTF16 (never happens, previous case would trigger first)
    - command cannot be converted to UTF16
    - the syscall itself failed
- Shell error, exit code > 255:
    - something failed after the syscall and before the linux proces started
- Shell error, exit code < 256:
    - linux proces returned non-zero exit code
    - something failed Windows-side after the linux process started
        - WSL is shut down (exit code 1)
        - The distro is unregistered (exit code 1)
        - etc?

I'm working on it at the moment, but it's not so simple as Windows and Linux errors.

Reply:
Seeing that we cannot guarantee that an error comes form linux, I think we should stick to the current design

Reply:
agreed